### DISCUSSION TITLE
setting charge period 2 times to Octopus intelligent time slots

### DISCUSSION THREAD

**Discussion by Alllfff** [nathanmarlor/foxess_modbus_charge_period_card#11] [Q&A] [answered]
I would like an automation to take the start and end time stamps of the "off peak" slot from Octopus Intelligent HA integration and fill them in the times of the "charge period 2" times in this integration, any way to do this please? Thank you

On another note, I can set the work mode of the Foxess to "Back up" by an automation which will prevent the EV draining the home battery but this will not allow to cheaply charge the home battery at the same time.

**Comment by canton7** [comment, marked as answer]
> I would like an automation to take the start and end time stamps of the "off peak" slot from Octopus Intelligent HA integration and fill them in the times of the "charge period 2" times in this integration, any way to do this please? Thank you

You'd do this by calling a service from the foxess_modbus integration, not by using the charge card (which is just UI). There's a service you can call to set a charge period, and pass the start/end times from the Octopus integration. I think https://github.com/nathanmarlor/foxess_modbus/discussions/404 was doing this?

> On another note, I can set the work mode of the Foxess to "Back up" by an automation which will prevent the EV draining the home battery but this will not allow to cheaply charge the home battery at the same time.

So, set a charge period for this slot, which will prevent discharge as well as charging the battery?

**Reply by Alllfff** [reply]
> So, set a charge period for this slot, which will prevent discharge as well as charging the battery?

yes this is the plan A

> I think https://github.com/nathanmarlor/foxess_modbus/discussions/404 was doing this?

thank you, I will give it a go

**Comment by Alllfff** [comment]
@canton7 It just occurred to me I could set the second period on for full day and only work with the enable period toggle when required.

**Reply by canton7** [reply]
Yep, that should work. Or set the end time for the next hour or half-our (so if `now` is 4.15, set the end time at 4.30 or 5.00)
